src/Juvix/Backends/LLVM/Net/API.hs

```diff
@@ -96,56 +108,63 @@ defineAppendToNet =
-ptr ← Codegen.loadElementPtr (Codegen.Minimal {Codegen.address' = nodes, Codegen.type' = nodePointer, Codegen.indincies' = [ind, Operand.ConstantOperand (C.Int 32 0)]})
--- TODO: Link things, lookup node pointers.
--- Alter parameter?
-next ← Codegen.add int32 ind (Operand.ConstantOperand (C.Int 32 1))
-Codegen.store counter next
-cond ← Codegen.icmp IntPred.EQ node_count next
-Codegen.cbr cond forLoop forExit
--- Exit case: return.
-Codegen.setBlock forExit2
--- TODO: Set eac list pointer?
-Codegen.retNull
--}
+  where
+    args = [(opaqueNetType, "net"), (nodePointer, "nodes"), (int32, "node_count")]
+
+defineAppendToNet' ∷ (Codegen.Define m, Codegen.MallocNode m) ⇒ m Operand.Operand
+defineAppendToNet' =
+  Codegen.defineFunction Type.void "append_to_net" args $ do
+    nodes ← Codegen.externf "nodes"
+    node_count ← Codegen.externf "node_count"
+    forLoop ← Codegen.addBlock "for.loop"
+    forExit ← Codegen.addBlock "for.exit"
+    forLoop2 ← Codegen.addBlock "for.loop.2"
+    forExit2 ← Codegen.addBlock "for.exit.2"
+    -- Create a counter to track position
+    counter ← Codegen.alloca int32
+    Codegen.store counter (Operand.ConstantOperand (C.Int 32 0))
+    Codegen.br forLoop
+    -- Loop case: convert node, increment counter.
+    Codegen.setBlock forLoop
+    ind ← Codegen.load int32 counter
+    -- Load node at index `ind`.
+    node ← Codegen.loadElementPtr
+      (Codegen.Minimal {Codegen.address' = nodes, Codegen.type' = nodePointer, Codegen.indincies' = [ind]})
+    -- Create the in-memory node.
+    kind ← EAC.mallocApp -- TODO: Switch on node kind.
+      -- Write the address to the list.
+    addr ← Codegen.getElementPtr
+      (Codegen.Minimal {Codegen.address' = nodes, Codegen.type' = nodePointer, Codegen.indincies' = [ind, Operand.ConstantOperand (C.Int 32 0)]})
+    Codegen.store addr kind
+    next ← Codegen.add int32 ind (Operand.ConstantOperand (C.Int 32 1))
+    Codegen.store counter next
+    cond ← Codegen.icmp IntPred.EQ node_count next
+    Codegen.cbr cond forLoop forExit
+    -- Exit case: next loop.
+    Codegen.setBlock forExit
+    Codegen.retNull
+    Codegen.store counter (Operand.ConstantOperand (C.Int 32 0))
+    Codegen.br forLoop2
+    -- Second loop: link nodes.
+    Codegen.setBlock forLoop2
+    ind ← Codegen.load int32 counter
+    -- Load node at index `ind`.
+    node ← Codegen.loadElementPtr
+      (Codegen.Minimal {Codegen.address' = nodes, Codegen.type' = nodePointer, Codegen.indincies' = [ind]})
+    ptr ← Codegen.loadElementPtr
+      (Codegen.Minimal {Codegen.address' = nodes, Codegen.type' = nodePointer, Codegen.indincies' = [ind, Operand.ConstantOperand (C.Int 32 0)]})
+    -- TODO: Link things, lookup node pointers.
+    -- Alter parameter?
+    next ← Codegen.add int32 ind (Operand.ConstantOperand (C.Int 32 1))
+    Codegen.store counter next
+    cond ← Codegen.icmp IntPred.EQ node_count next
+    Codegen.cbr cond forLoop forExit
```

Comment:
`    Codegen.cbr cond forLoop2 forExit2`

this is probably what you wanted, with this, your increment would be correct